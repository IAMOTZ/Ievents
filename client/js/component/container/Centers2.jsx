import React from 'react';
import { connect } from 'react-redux';
import { Rediret } from 'react-router-dom';
import getAllCenters from '../../actions/centerActions';
import { showCenterModal } from '../../actions/centerActions';

import styles from '../../../sass/centers2.scss';
import { UserTopNav } from '../common/TopNavigation.jsx';
import UserSideNav from '../common/SideNavigation.jsx';
import Header from '../common/Header.jsx';
import CenterCards from '../common/CenterCards.jsx';
import { CenterModal } from '../common/Modal';


@connect((store) => {
  return {
    authenticated: store.user.status.fetched,
    user: store.user.user,
    centers: store.centers.centers,
    modalContent: store.centers.modalContent ? store.centers.modalContent : {
      name: null,
      images: [],
      details: null,
      capacity: null,
      price: null,
      bookedOn: [],
      type: null,
    },
    isAdmin: (store.user.user.role === 'admin') ? true : false,
  }
})

export default class Centers2 extends React.Component {
  componentWillMount() {
    this.props.dispatch(getAllCenters());
  }

  showModal = (e) => {
    this.props.dispatch(showCenterModal(e.target.id));
  }

  render() {
    if (!this.props.authenticated) {
      return (<Redirect to="/users/login" />);
    } else {
      return (
        <div>
          {/* Top Navigarion on Small screen */}
          <UserTopNav name={this.props.user.name} title="Centers" />

          <div className="container-fluid">
            <div className="row">

              {/* Side Navigation on large screen */}
              <UserSideNav userName={this.props.user.name} />

              {/* Main content */}
              <div class="col-lg-10 offset-lg-2" id="centers-section">
                {/* Content Headers */}
                <Header text="Centers" />

                <section id="centers-section">
                  <div className="container">
                    {/* Search Box */}
                    <div id="center-section-content" class="d-flex flex-column align-items-center">
                      <div class="input-group w-50 mt-lg-5 search-box">
                        <input type="text" class="form-control" placeholder="Search for a center" />
                        <div class="input-group-btn d-none d-sm-block">
                          <button type="button" class="btn btn-secondary dropdown-toggle" data-toggle="dropdown">
                            Search with
                          </button>
                          <div class="dropdown-menu dropdown-menu-right">
                            <a class="dropdown-item" href="#">Location</a>
                            <div role="separator" class="dropdown-divider"></div>
                            <a class="dropdown-item" href="#">Name</a>
                          </div>
                        </div>
                      </div>
                      <div class="search-box mt-3 d-block d-sm-none">
                        <button type="button" class="btn btn-secondary dropdown-toggle" data-toggle="dropdown">
                          Search with
                        </button>
                        <div class="dropdown-menu">
                          <a class="dropdown-item" href="#">Location</a>
                          <div role="separator" class="dropdown-divider"></div>
                          <a class="dropdown-item" href="#">Name</a>
                        </div>
                      </div>
                    </div>

                    {/* Centers  Grid*/}
                    <div className="mt-5">
                      <div className="row">
                        <CenterCards centers={this.props.centers} btnAction={this.showModal} isAdmin={this.props.isAdmin} />
                      </div>
                    </div>
                  </div>
                </section>

                {/* Modal */}
                <CenterModal modalContent={this.props.modalContent} />

                {/* Footer on large screen */}
                <footer class="d-none d-lg-block mt-5">
                  <div class="container text-white text-center py-5">
                    <h1>Ievents</h1>
                    <p>Copyright &copy; 2017</p>
                  </div>
                </footer>
              </div>
            </div>
          </div>

          {/* Footer on small screen */}
          <footer class="d-block d-lg-none mt-5">
            <div class="container text-white text-center py-5">
              <h1>Ievents</h1>
              <p>Copyright &copy; 2017</p>
            </div>
          </footer>
        </div>

      )
    }

  }
}