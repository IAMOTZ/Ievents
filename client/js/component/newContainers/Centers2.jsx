import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { find } from 'lodash';
import {
  getAllCenters, setCenterToUpdate,
} from '../../actions/centerActions';
import { setCenterToBook } from '../../actions/eventActions';
import UserSideNav from '../common/SideNavigation';
import Header from '../common/Header';
import CenterCards from '../common/CenterCards';
import { CenterModal } from '../common/Modal';
import { LoadingContainer } from '../common/LoadingAnimation';
import { UserTopNav } from '../common/TopNavigation';

@connect((store) => {
  const { user } = store.authReducer;
  return {
    user,
    isAdmin: user.role === 'admin',
    isSuperAdmin: user.role === 'admin' || user.role === 'superAdmin',
    isUserAuthenticated: store.authReducer.loggingUserResolved,
    fetchingCentersStarted: store.fetchCentersReducer.fetchingCenterStarted,
    centers: store.fetchCentersReducer.centers,
  };
})
class Centers2 extends React.Component {
  constructor() {
    super();
    this.state = {
      modalContent: null,
    };
  }

  componentWillMount() {
    this.props.dispatch(getAllCenters());
  }

  /**
   * It updates the store about a center that is to be edited.
   * @param {Event} event The event object.
   */
  onEdit = (event) => {
    this.props.dispatch(setCenterToUpdate(event.target.id));
  }

  /**
   * It updates the store about a center that is to be booked.
   * @param {Number} centerId The ID of the center.
   */
  onBook = (centerId) => {
    this.props.dispatch(setCenterToBook(centerId));
  }

  /**
   * Displays the center modal.
   * It uses the center ID to get the details of the center to show.
   * @param {Event} event The event object.
   */
  showModal = (event) => {
    const state = { ...this.state };
    const centerId = Number(event.target.id);
    state.modalContent = find(this.props.centers, { id: centerId });
    this.setState(state);
  }

  render() {
    if (!this.props.isUserAuthenticated) {
      return (<Redirect to="/users/login" />);
    }
    return (
      <div id="centers2-container">
        {/* Top Navigarion on Small screen */}
        <UserTopNav
          name={this.props.user.name}
          title="Centers"
          isAdmin={this.props.isAdmin}
          isSuperAdmin={this.props.isSuperAdmin}
          dispatch={this.props.dispatch}
        />
        <div className="container-fluid">
          <div className="row">
            {/* Side Navigation on large screen */}
            <UserSideNav
              name={this.props.user.name}
              isAdmin={this.props.isAdmin}
              isSuperAdmin={this.props.isSuperAdmin}
              dispatch={this.props.dispatch}
            />
            {/* Main content */}
            <div className="col-lg-10 offset-lg-2 mt-lg-0" id="main-content">
              {/* Content Headers */}
              <Header text="Centers" />
              {
                this.props.centers.length === 0 && this.props.fetchingCentersStarted ?
                  <LoadingContainer iconSize={4} /> :
                  <section id="centers-section">
                    <div className="container">
                      {/* Search Box */}
                      <div
                        id="center-section-content"
                        className="d-flex flex-column align-items-center"
                      >
                        <div className="input-group w-50 mt-lg-5 search-box">
                          <input
                            type="text"
                            className="form-control"
                            placeholder="Search for a center"
                          />
                          <div className="input-group-btn d-none d-sm-block">
                            <button
                              type="button"
                              className="btn btn-secondary dropdown-toggle"
                              data-toggle="dropdown"
                            >Search with
                              </button>
                            <div className="dropdown-menu dropdown-menu-right">
                              <a className="dropdown-item" href="#">Location</a>
                              <div role="separator" className="dropdown-divider" />
                              <a className="dropdown-item" href="#">Name</a>
                            </div>
                          </div>
                        </div>
                        <div className="search-box mt-3 d-block d-sm-none">
                          <button
                            type="button"
                            className="btn btn-secondary dropdown-toggle"
                            data-toggle="dropdown"
                          >Search with
                            </button>
                          <div className="dropdown-menu">
                            <a className="dropdown-item" href="#">Location</a>
                            <div role="separator" className="dropdown-divider" />
                            <a className="dropdown-item" href="#">Name</a>
                          </div>
                        </div>
                      </div>
                      {/* Centers  Grid */}
                      <div className="mt-5">
                        <div className="row">
                          <CenterCards
                            centers={this.props.centers}
                            btnAction={this.showModal}
                            editAction={this.onEdit}
                            isAdmin={this.props.isAdmin || this.props.isSuperAdmin}
                          />
                        </div>
                      </div>
                    </div>
                  </section>
              }
              {/* Modal */}
              {
                this.state.modalContent ? <CenterModal
                  modalContent={this.state.modalContent}
                  onBook={this.onBook}
                  redirectPath="/users/login"
                /> : null
              }
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Centers2;
