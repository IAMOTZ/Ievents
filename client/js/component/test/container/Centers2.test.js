/* global shallow */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import Centers2 from '../../container/Centers2';

const props = {
  authenticated: true,
  user: { name: 'test-user' },
  centers: [
    { id: 1, name: 'test-center1' },
    { id: 2, name: 'test-center2' },
  ],
  modalContent: {
    name: null,
    images: [],
    details: null,
    capacity: null,
    price: null,
    bookedOn: [],
    type: null,
  },
  isAdmin: true,
  isSuperAdmin: false,
  status: {
    fetching: false,
  },
  dispatch: () => { },
};

const alterProps = newProps => ({
  ...props,
  centers: newProps.centers || props.centers,
  authenticated: newProps.authenticated === undefined ? true : newProps.authenticated,
  status: {
    fetching: newProps.fetching || false,
  },
});

describe('<Centers2 />', () => {
  describe('Rendering:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Centers2 {...props} />);
    })
    it('should render well on page load', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should redirect to login page is user is not authenticated', () => {
      wrapper.setProps(alterProps({ authenticated: false }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should render well when fetching centers', () => {
      wrapper.setProps(alterProps({ fetching: true, centers: [] }));
      expect(wrapper).toMatchSnapshot();
    });
  });
});
