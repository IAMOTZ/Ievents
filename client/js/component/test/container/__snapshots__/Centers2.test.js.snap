// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Centers2 /> Rendering: should redirect to login page is user is not authenticated 1`] = `
<Redirect
  push={false}
  to="/users/login"
/>
`;

exports[`<Centers2 /> Rendering: should render well on page load 1`] = `
<div
  id="centers2-container"
>
  <UserTopNav
    dispatch={[Function]}
    isAdmin={true}
    isSuperAdmin={false}
    name="test-user"
    title="Centers"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <UserSideNav
        dispatch={[Function]}
        isAdmin={true}
        isSuperAdmin={false}
        name="test-user"
      />
      <div
        className="col-lg-10 offset-lg-2 mt-lg-0"
        id="main-content"
      >
        <Header
          text="Centers"
        />
        <section
          id="centers-section"
        >
          <div
            className="container"
          >
            <div
              className="d-flex flex-column align-items-center"
              id="center-section-content"
            >
              <div
                className="input-group w-50 mt-lg-5 search-box"
              >
                <input
                  className="form-control"
                  placeholder="Search for a center"
                  type="text"
                />
                <div
                  className="input-group-btn d-none d-sm-block"
                >
                  <button
                    className="btn btn-secondary dropdown-toggle"
                    data-toggle="dropdown"
                    type="button"
                  >
                    Search with
                  </button>
                  <div
                    className="dropdown-menu dropdown-menu-right"
                  >
                    <a
                      className="dropdown-item"
                      href="#"
                    >
                      Location
                    </a>
                    <div
                      className="dropdown-divider"
                      role="separator"
                    />
                    <a
                      className="dropdown-item"
                      href="#"
                    >
                      Name
                    </a>
                  </div>
                </div>
              </div>
              <div
                className="search-box mt-3 d-block d-sm-none"
              >
                <button
                  className="btn btn-secondary dropdown-toggle"
                  data-toggle="dropdown"
                  type="button"
                >
                  Search with
                </button>
                <div
                  className="dropdown-menu"
                >
                  <a
                    className="dropdown-item"
                    href="#"
                  >
                    Location
                  </a>
                  <div
                    className="dropdown-divider"
                    role="separator"
                  />
                  <a
                    className="dropdown-item"
                    href="#"
                  >
                    Name
                  </a>
                </div>
              </div>
            </div>
            <div
              className="mt-5"
            >
              <div
                className="row"
              >
                <CenterCards
                  btnAction={[Function]}
                  centers={
                    Array [
                      Object {
                        "id": 1,
                        "name": "test-center1",
                      },
                      Object {
                        "id": 2,
                        "name": "test-center2",
                      },
                    ]
                  }
                  editAction={[Function]}
                  isAdmin={true}
                />
              </div>
            </div>
          </div>
        </section>
        <CenterModal
          modalContent={
            Object {
              "bookedOn": Array [],
              "capacity": null,
              "details": null,
              "images": Array [],
              "name": null,
              "price": null,
              "type": null,
            }
          }
          onBook={[Function]}
          redirectPath="/addEvent"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Centers2 /> Rendering: should render well when fetching centers 1`] = `
<div
  id="centers2-container"
>
  <UserTopNav
    dispatch={[Function]}
    isAdmin={true}
    isSuperAdmin={false}
    name="test-user"
    title="Centers"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <UserSideNav
        dispatch={[Function]}
        isAdmin={true}
        isSuperAdmin={false}
        name="test-user"
      />
      <div
        className="col-lg-10 offset-lg-2 mt-lg-0"
        id="main-content"
      >
        <Header
          text="Centers"
        />
        <LoadingContainer
          iconSize={4}
        />
        <CenterModal
          modalContent={
            Object {
              "bookedOn": Array [],
              "capacity": null,
              "details": null,
              "images": Array [],
              "name": null,
              "price": null,
              "type": null,
            }
          }
          onBook={[Function]}
          redirectPath="/addEvent"
        />
      </div>
    </div>
  </div>
</div>
`;
